google/cloud/pubsub/samples/pubsub_samples_common.cc

```diff
@@ -167,25 +168,38 @@ std::string ReadFile(std::string const& path) {
   return std::string{std::istreambuf_iterator<char>{ifs.rdbuf()}, {}};
 }
 
-std::string CommitSchemaRevisionsForRollbackSchemaTesting(
-    google::cloud::pubsub::SchemaServiceClient& schema_admin,
+std::pair<std::string, std::string> CommitSchemaWithRevisionsForTesting(
+    google::cloud::pubsub::SchemaServiceClient& client,
     std::string const& project_id, std::string const& schema_id,
-    std::string const& file) {
-  std::string const definition = ReadFile(file);
+    std::string const& schema_file, std::string const& revised_schema_file,
+    std::string const& type) {
+  std::string const initial_definition = ReadFile(schema_file);
+  std::string const revised_definition = ReadFile(revised_schema_file);
+  auto const schema_type = type == "AVRO"
+                               ? google::pubsub::v1::Schema::AVRO
+                               : google::pubsub::v1::Schema::PROTOCOL_BUFFER;
+
+  google::pubsub::v1::CreateSchemaRequest create_request;
+  create_request.set_parent(google::cloud::Project(project_id).FullName());
+  create_request.set_schema_id(schema_id);
+  create_request.mutable_schema()->set_type(schema_type);
+  create_request.mutable_schema()->set_definition(initial_definition);
+  auto schema = client.CreateSchema(create_request);
+  if (!schema) throw std::move(schema).status();
+  std::string const first_revision_id = schema.value().revision_id();
```

Comment:
We spell these `schema->revision_id()`.  Also, making this a `const` prevents move-from.  Does not matter in this test, but generally it would be more efficient to write:

```cc
... ..
  auto first_revision_id = schema->value();
...
  return {std::move(first_revision_id), std::move(last_revision_id)}
```

If you want to save even more copying you would write:

```cc
  auto first_revision_id = std::move(*schema->mutable_revision_id());
 ..
...
  return {std::move(first_revision_id), std::move(*schema->mutable_revision_id())};
```

Reply:
Ack. For StatusOr's, use indirection instead of .value().

Changed so that we are using std::move in the return. I think the second version is overkill for the samples.